docs/basic_training/processes.pt.md

```diff
@@ -0,0 +1,950 @@
+
+```console
+tarefa 3 do processo
+tarefa 1 do processo
+tarefa 2 do processo
+```
+
+!!! warning
+
+    O canal garante que os itens sejam entregues na mesma ordem em que foram enviados - mas - como o processo é executado de forma paralela, não há garantia de que sejam processados na mesma ordem em que foram recebidos.
+
+### Arquivo e caminhos de entrada
+
+O qualificador `path` permite a manipulação de arquivos no contexto de execução do processo. Isso significa que o Nextflow irá mover os arquivos necessários para o diretório de execução do processo e estes poderão ser acessados no script usando o nome especificado na declaração de entrada.
+
+```groovy linenums="1"
+leituras = Channel.fromPath( 'data/ggal/*.fq' )
+
+process FOO {
+  debug true
+
+  input:
+  path 'amostra.fastq'
+
+  script:
+  """
+  ls amostra.fastq
+  """
+}
+
+workflow {
+  resultado = FOO(leituras)
+}
+```
+
+O nome do arquivo de entrada também pode ser definido usando uma referência de variável conforme mostrado abaixo:
+
+```groovy linenums="1"
+leituras = Channel.fromPath( 'data/ggal/*.fq' )
+
+process FOO {
+  debug true
+
+  input:
+  path amostra
+
+  script:
+  """
+  ls  $amostra
+  """
+}
+
+workflow {
+  resultado = FOO(leituras)
+}
+```
+
+A mesma sintaxe também é capaz de lidar com mais de um arquivo de entrada na mesma execução e requer apenas a alteração da composição do canal.
+
+```groovy linenums="1"
+leituras = Channel.fromPath( 'data/ggal/*.fq' )
+
+process FOO {
+  debug true
+
+  input:
+  path amostra
+
+  script:
+  """
+  ls -lh $amostra
+  """
+}
+
+workflow {
+  FOO(leituras.collect())
+}
+```
+
+!!! warning
+
+    No passado, o qualificador `file` era usado para arquivos, mas o qualificador `path` deve ser preferido ao `file` para lidar com arquivos de entrada de processo ao usar o Nextflow 19.10.0 ou posterior. Quando um processo declara um arquivo de entrada, os elementos de canal correspondentes devem ser objetos **file** criados com a função auxiliar de arquivo das fábricas de canal específicas de arquivo (por exemplo, `Channel.fromPath` ou `Channel.fromFilePairs`).
+
+!!! exercise
+
+    Escreva um script que crie um canal contendo todos as leituras correspondentes ao padrão `data/ggal/*_1.fq` seguido por um processo que os concatene em um único arquivo e imprima as primeiras 20 linhas.
+
+
+    ??? solution
+
+        ```groovy linenums="1"
+        params.leituras = "$baseDir/data/ggal/*_1.fq"
+
+        Channel
+          .fromPath( params.leituras )
+          .set { canal_leituras }
+
+        process CONCATENE {
+          tag "Concatene todos os arquivos"
+
+          input:
+          path '*'
+
+          output:
+          path 'top_10_linhas'
+
+          script:
+          """
+          cat * > concatenado.txt
+          head -n 20 concatenado.txt > top_10_linhas
```

Comment:
If it's `head -n 20` shouldn't it be top_20_linhas?

Reply:
Great point 😆 I have written down to fix this on the original material in English. Once fixed there, I will come to fix this here.